mne/tests/test_cov.py

```diff
@@ -75,6 +76,24 @@ def test_compute_whitener(proj, pca):
         assert pca is False
         assert_allclose(round_trip, np.eye(n_channels), atol=0.05)
 
+    raw.info['bads'] = [raw.ch_names[0]]
+    picks = pick_types(raw.info, meg=True, eeg=True, exclude=[])
+    with pytest.warns(RuntimeWarning, match='Too few samples'):
+        cov2 = compute_raw_covariance(raw, picks=picks)
+        cov3 = compute_raw_covariance(raw, picks=None)
+    assert_array_equal(cov2['data'][1:, 1:], cov3['data'])
```

Comment:
With this `assert_array_equal` we should get some handle on how many entries are actually wrong. It could be a numerical imprecision issue, in which case `assert_allclose` should fix it. For example, accumulators using SIMD instructions could differ slightly and result in differences around epsilon, and `assert_allclose`'s `rtol` should take care of this (as long as the values are not zero-like, in which case we'd have to set `atol` properly -- but hopefully this is not the issue.)

Let's first see what the failing Azure CI says here, then hopefully committing this will fix the issue:

```suggestion
    assert_allclose(cov2['data'][1:, 1:], cov3['data'])
```

Reply:
I was guessing the problem was in those array equality assertions, but I didn't know about both of those convenient functions. Let's see the result, thanks!

Reply:
Quoth GitHub Actions:
```
Max relative difference: 1.93161015e-12
```
So this should indeed be fixed by assert_allclose, which defaults to `rtol=1e-7`. Let's find out!